plugins/woocommerce/client/legacy/js/frontend/add-to-cart.js

```diff
@@ -95,7 +109,7 @@ jQuery( function( $ ) {
 			e.data.addToCartHandler.addRequest({
 				type: 'POST',
 				url: wc_add_to_cart_params.wc_ajax_url.toString().replace( '%%endpoint%%', 'add_to_cart' ),
-				data: data,
+				data: $.param( data ),
```

Comment:
@apenchev We can remove JQuery here in favor of `URLSearchParams`.

I think this should work:

```suggestion
				data: new URLSearchParams( data ).toString(),
```

Reply:
Thanks for checking! URLSearchParams() can't handle nested data - I _believe_ the only file with nested data is **checkout.js**:

```
var shipping_methods = {};
$( 'select.shipping_method, input[name^="shipping_method"][type="radio"]:checked, input[name^="shipping_method"][type="hidden"]' ).each( function() {
  shipping_methods[ $( this ).data( 'index' ) ] = $( this ).val();
} );
data.shipping_method = shipping_methods;
```

So it should be fine to use URLSearchParams() here, in **add-to-cart.js**.

Reply:
@apenchev what does $.params do with nested data? just flatten it as a series of query params or something else? Because we could just flatten all the object props out and then pass to URLSearchParams?

Reply:
I figured out how to flatten **shipping_method**. However, I have no idea how to test **add-to-cart-variations.js** and if it contains nested data or not. That's the last file in the PR using $.param(), the rest should be good now.